planner/core/find_best_task.go

```diff
@@ -1281,7 +1288,9 @@ func matchIndicesProp(idxCols []*expression.Column, colLens []int, propItems []p
 		return false
 	}
 	for i, item := range propItems {
-		if colLens[i] != types.UnspecifiedLength || !item.Col.Equal(nil, idxCols[i]) {
+		expr, isOk := item.Col.VirtualExpr.(*expression.ScalarFunction)
```

Comment:
Can we use a function to wrap these code?

Reply:
Done in the new commit